### PR TITLE
Support TiDB Cloud Telemetry

### LINKED ISSUE #19506: Support TiDB Cloud Telemetry
**Is your feature request related to a problem? Please describe.**
Currently, we do not configure the telemetry tag. So we can not know if the user used the DBeaver to link to the TiDB.

**Describe the solution you'd like**
Add `connectionAttributes=program_name:dbeaver_tidb_plugin` property as the default driver property to mark it.

### PR DESCRIPTION
**Pull request by Icemap**
Added default property to mark the program name.

### CODE REVIEW

**Comment by Icemap** [comment]
Refers to https://github.com/dbeaver/dbeaver/issues/19506

**Comment by LonwoLonwo** [comment]
Hello @Icemap 

Thanks for your contribution.

But I am not sure that this is the best way to send this value.

Please take a look at this code:

https://github.com/dbeaver/dbeaver/blob/267ebf5a324189d2a16d8250340398344229dbe7/plugins/org.jkiss.dbeaver.ext.hana/src/org/jkiss/dbeaver/ext/hana/model/HANADataSource.java#L115-L118

**Comment by Icemap** [comment]
OK, I'll rewrite it! THX!

**Comment by Icemap** [comment]
@LonwoLonwo Changed it to use the code way to set inner properties. PTAL.

**Comment by LonwoLonwo** [comment]
Please add the result of the DBUtils.getClientApplicationName method in the value as well.

**Comment by Icemap** [comment]
Added.

File: plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/mysql/model/TiDBMySQLDataSource.java [hunk 1]
```diff
@@ -145,4 +152,19 @@ public boolean isServerVersionAtLeast(int major, int minor) {
...
+    @Override
+    protected Map<String, String> getInternalConnectionProperties(DBRProgressMonitor monitor, DBPDriver driver, JDBCExecutionContext context, String purpose, DBPConnectionConfiguration connectionInfo) throws DBCException {
+        Map<String, String> props = super.getInternalConnectionProperties(monitor, driver, context, purpose, connectionInfo);
+        // build application name
+        String appName = DBUtils.getClientApplicationName(getContainer(), context, purpose);
+        appName = "dbeaver_tidb_plugin" + (appName.isEmpty() ? "" : "(" + appName + ")");
+        
+        // build conAttr value
+        String connAttr = props.get(CONN_ATTR_NAME);
+        connAttr = PROP_APPLICATION_NAME + ":" + appName + (connAttr == null || connAttr.isEmpty() ? "" : "," + connAttr);
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/mysql/model/TiDBMySQLDataSource.java]:
```suggestion
        connAttr = PROP_APPLICATION_NAME + ":" + appName + (CommonUtils.isEmpty(connAttr) ? "" : "," + connAttr);
```

+add CommonUtils import

**Comment by LonwoLonwo** [comment]
Do you have some advice on how to test it with TiDB?

File: plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/mysql/model/TiDBMySQLDataSource.java (hunk 1 above)
Comment (Icemap) [reply plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/mysql/model/TiDBMySQLDataSource.java]: Added.

**Comment by Icemap** [comment]
> Do you have some advice on how to test it with TiDB?

Yeah, I already test it in TiDB Cloud. We have a telemetry dashboard to watch it conveniently.
image

If you want to test it yourselves, you can create a [TiDB Cloud Serverless Tier Cluster](https://tidbcloud.com/) and use DBeaver to establish a connection. Then, tell me the cluster region. I can show you the result of the telemetry dashboard here.

**Review by LonwoLonwo — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

**Comment by uslss** [comment]
verified

**Comment by Icemap** [comment]
Sorry guys, we found a typo. It should be `"program_name"`.

https://github.com/dbeaver/dbeaver/blob/a7dad6ff2cf2f97823365616adc4e23a3f8c2c86/plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/mysql/model/TiDBMySQLDataSource.java#L56

In the test period, the `connectionAttributes` also can work, because I forgot to clean the config in driver properties.
image

I'll update it.

**Comment by Icemap** [comment]
This is typo fixed PR https://github.com/dbeaver/dbeaver/pull/19745. Sorry again.

**Comment by z87506386** [comment]
你好,我想了解一下如何通过dbeaver连接tidb,这是我目前遇到的问题Connections using insecure transport are prohibited. See https://docs.pingcap.com/tidbcloud/secure-connections-to-serverless-tier-clusters.是否因为我没有正确引入CA证书的原因

**Comment by Icemap** [comment]
@z87506386 
Hi, It's not your fault. That is because of this issue https://github.com/dbeaver/dbeaver/issues/18188.
你好，这不是你的错误导致的。这是由于这个问题 https://github.com/dbeaver/dbeaver/issues/18188 所导致的。

You can work around this bug by selecting `SSL` tab like this picture below.
你可以通过将 `SSL` 标签像下图一样选择从而暂时绕过这个 Bug。

image

**Pull request merged**